src/lib.rs

```diff
@@ -1,75 +1,77 @@
+use log::LevelFilter;
+use std::path::PathBuf;
 
+/// Command line arguments struct
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
 pub struct Args {
     #[arg(short = 'c', long, help = "Sets a custom config file")]
-    config: Option<std::path::PathBuf>,
+    config: Option<PathBuf>,
 
     #[arg(short = 'l', long, help = "Sets logging destination")]
-    log: Option<std::path::PathBuf>,
+    log: Option<PathBuf>,
 
     #[command(flatten)]
     verbose: Verbosity,
 }
 
+/// Configures the application using the parsed command line arguments
 pub fn configure(parsed_args: Args) -> Result<Config> {
     // Set defaults
-    let mut builder = Config::builder();
-    builder = builder
-        .set_default("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))
-        .context(ErrorKind::ConfigImmutable)?;
-    builder = builder
-        .set_default("Log.file", "_CONSOLE_")
-        .context(ErrorKind::ConfigImmutable)?;
-    builder = builder
-        .set_default("Web.address", "127.0.0.1")
-        .context(ErrorKind::ConfigImmutable)?;
-    builder = builder
-        .set_default("Web.port", 7878)
-        .context(ErrorKind::ConfigImmutable)?;
+    let mut builder = Config::builder()
+        .set_default("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))?
+        .set_default("Log.file", "_CONSOLE_")?
+        .set_default("Web.address", "127.0.0.1")?
+        .set_default("Web.port", 7878)?
+        .set_default("Database.type", "postgres")?;
 
     // Determine config file
-    let config_path = parsed_args.config.unwrap();
-    let config_file = config_path.to_str().unwrap();
+    let config_path = parsed_args.config.ok_or_else(|| anyhow::anyhow!("No config path provided"))?;
+    let config_file = config_path.to_str().ok_or_else(|| anyhow::anyhow!("Failed to convert config path to string"))?;
 
+    builder = set_config_file_override(builder, config_file)?;
+    builder = builder.add_source(File::new(config_file, FileFormat::Toml));
     builder = builder
-        .set_override("cfg_file", config_file)
+        .merge(create_override_config(config_file)?)
         .context(ErrorKind::ConfigImmutable)?;
 
-    builder = builder.add_source(File::new(config_file, FileFormat::Toml)); // Or whatever format
-
-        builder = builder
-            .set_override("cfg_file", config_file)
-            .context(ErrorKind::ConfigImmutable)?;
-
-    // Apply environmenet variable overrides
-    let env_vars = Environment::with_prefix("AARDWOLF")
+    // Apply environment variable overrides
+    let env_prefix = env!("CARGO_PKG_NAME").to_ascii_uppercase().replace_char('-', '_');
+    let env_vars = Environment::with_prefix(env_prefix)
         .separator("_")
         .ignore_empty(true);
     builder = builder.add_source(env_vars);
 
-    let log_path = parsed_args.log.unwrap(); 
+    let log_path = parsed_args.log.ok_or_else(|| anyhow::anyhow!("No log path provided"))?;
```

Comment:
**Suggestion:** In the `configure` function, you are unwrapping the `config` and `log` options without checking if they are `None`. This could lead to a panic if they are `None`. Instead, you should provide a default value or handle the `None` case properly.
```suggestion
    let config_path = parsed_args.config.unwrap_or_else(|| PathBuf::from("default_config_path"));
    let log_path = parsed_args.log.unwrap_or_else(|| PathBuf::from("default_log_path"));
```